x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/watch/WatchParser.java

```diff
@@ -174,6 +175,13 @@ public Watch parse(String id, boolean includeStatus, WatcherXContentParser parse
                 throw new ElasticsearchParseException("could not parse watch [{}]. unexpected field [{}]", id, currentFieldName);
             }
         }
+        /* Make sure we are at the end of the available input data -- certain types of JSON errors will not manifest
```

Comment:
minor nit: can you use a `//` comment like the other comments in the methods as well and add new lines before and after?